### PR TITLE
Split database interfaces in seperate files

### PR DESCRIPTION
**Pull request by ghost**
- Split ilDBInterface and ilDBPdoInterface in seperate files
- Split ilDBManager and ilDBPdoManagerInterface in seperate files
- Correct name ilQueryUtilsInterface file name

### CODE REVIEW

File: Services/Database/classes/QueryUtils/class.ilQueryUtils.php [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?php
-require_once 'Services/Database/interfaces/interface.ilQueryUtils.php';
+require_once 'Services/Database/interfaces/interface.ilQueryUtilsInterface.php';
```
Comment (chfsx) [inline Services/Database/classes/QueryUtils/class.ilQueryUtils.php]: @sragfw you can remove the require completely since we now have autoloading.

**Review by chfsx — CHANGES_REQUESTED** [review]
Thanks for splitting them up!

**Comment by ghost** [comment]
@chfsx Removed alle require/include's inside database service and all requires in terms of database outside database service

**Comment by ghost** [comment]
@chfsx 
Can I close this PR since you have cherry-picked some parts yourself?

**Comment by chfsx** [comment]
I cherry-picked now all changes (and reverted some in tests) except the changes in the db-update script. thanks for splitting these interfaces up. I close the PR now

**Pull request closed without merging**